uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -839,6 +838,162 @@ public function testGetLeavePeriodEntitlementRemainderWithContactAndPeriodId() {
+    $this->createExpiredBroughtForwardBalanceChange($periodEntitlement->id, 8, 3);
+    $breakdowns = $periodEntitlement->getBreakdownBalanceChanges(false);
+
+    //there should only be three leave balance changes for this period entitlement that are not expired
+    $this->assertCount(3, $breakdowns);
+    $this->assertInstanceOf(LeaveBalanceChange::class, $breakdowns[0]);
+    $this->assertInstanceOf(LeaveBalanceChange::class, $breakdowns[1]);
+    $this->assertInstanceOf(LeaveBalanceChange::class, $breakdowns[2]);
+    $this->assertEquals($periodEntitlement->id, $breakdowns[0]->source_id);
+    $this->assertEquals(10, $breakdowns[0]->amount);
+    $this->assertEquals('entitlement', $breakdowns[0]->source_type);
+    $this->assertEquals($periodEntitlement->id, $breakdowns[1]->source_id);
+    $this->assertEquals(9, $breakdowns[1]->amount);
+    $this->assertEquals('entitlement', $breakdowns[1]->source_type);
+    $this->assertEquals($periodEntitlement->id, $breakdowns[2]->source_id);
+    $this->assertEquals(8, $breakdowns[2]->amount);
+    $this->assertEquals('entitlement', $breakdowns[2]->source_type);
+  }
+
+  public function testgetBreakdownBalanceChangesShouldIncludeOnlyExpiredBalancesWhenTrueIsPassed() {
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate();
+    $this->createLeaveBalanceChange($periodEntitlement->id, 10);
+    $this->createExpiredBroughtForwardBalanceChange($periodEntitlement->id, 9, 5);
+    $this->createExpiredBroughtForwardBalanceChange($periodEntitlement->id, 8, 3);
+    $breakdowns = $periodEntitlement->getBreakdownBalanceChanges(true);
+
+    //there should only be two leave balance changes for this period entitlement that are expired
+    $this->assertCount(2, $breakdowns);
+    $this->assertInstanceOf(LeaveBalanceChange::class, $breakdowns[0]);
+    $this->assertInstanceOf(LeaveBalanceChange::class, $breakdowns[1]);
+    $this->assertEquals($periodEntitlement->id, $breakdowns[0]->source_id);
+    $this->assertEquals(-5, $breakdowns[0]->amount);
+    $this->assertEquals('entitlement', $breakdowns[0]->source_type);
+    $this->assertEquals($periodEntitlement->id, $breakdowns[1]->source_id);
+    $this->assertEquals(-3, $breakdowns[1]->amount);
+    $this->assertEquals('entitlement', $breakdowns[1]->source_type);
+  }
+
+  public function testGetLeavePeriodEntitlementBreakdown() {
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate();
+    $this->createExpiredBroughtForwardBalanceChange($periodEntitlement->id, 9, 5);
+    $this->createExpiredBroughtForwardBalanceChange($periodEntitlement->id, 8, 3);
+    $this->createLeaveBalanceChange($periodEntitlement->id, 10);
+
+    $params = ['entitlement_id' => $periodEntitlement->id];
+    $result = LeavePeriodEntitlement::getBreakdown($params);
+    $this->assertCount(1, $result);
+    $this->assertArrayHasKey('id', $result[0]);
+    $this->assertEquals($periodEntitlement->id, $result[0]['id']);
```

Comment:
What if result does not have the id key by any chance?
Actually had that issue and it threw an error , thats why I added that.